### PR TITLE
stm32g0 add dma & dmamux

### PR DESCRIPTION
**Pull request by grevaillot**
hi,

Here's a PR to enable dma on stm32g0 and add dmamux, a new peripheral needed to handle dma request peripheral mapping, found on a couple of new st chips.

### CODE REVIEW

File: include/libopencm3/stm32/common/dmamux_common_all.h [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+void dmamux_disable_request_generator(uint32_t dmamux, uint8_t rg_channel);
+
+void dmamux_set_request_generator_trigger(uint32_t dmamux, uint8_t rg_channel, uint8_t sig_id);
+void dmamux_set_request_generator_trigger_pol(uint32_t dmamux, uint8_t rg_channel, uint8_t polarity);
+void dmamux_set_request_generator_trigger_gnbreq(uint32_t dmamux, uint8_t rg_channel, uint8_t gnbreq);
+
+void dmamux_enable_request_generator_trigger_overrun_interrupt(uint32_t dmamux, uint8_t rg_channel);
+void dmamux_disable_request_generator_trigger_overrun_interrupt(uint32_t dmamux, uint8_t rg_channel);
+uint32_t dmamux_get_request_generator_trigger_overrun_interrupt(uint32_t dmamux, uint8_t rg_channel);
+void dmamux_clear_request_generator_trigger_overrun_interrupt(uint32_t dmamux, uint8_t rg_channel);
```
Comment (karlp) [inline include/libopencm3/stm32/common/dmamux_common_all.h]: wow. I mean, these are consistent namings, but wow :) getting long.

File: include/libopencm3/stm32/g0/dmamux.h [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+#define DMAMUX1				DMAMUX_BASE
+/**@}*/
+
+/* --- DMAMUX_CxCR values ------------------------------------ */
+
+/** @defgroup dmamux_cxcr_sync_id SYNCID Synchronization input selected
+  @ingroup dmamux_defines
+@{*/
+#define DMAMUX_CxCR_SYNC_ID_SYNC_EXTI_LINE0  0
+#define DMAMUX_CxCR_SYNC_ID_EXTI_LINE1       1
```
Comment (karlp) [inline include/libopencm3/stm32/g0/dmamux.h]: first one looks wrong?

Comment (grevaillot) [reply include/libopencm3/stm32/g0/dmamux.h]: thx, fixed.

File: include/libopencm3/stm32/common/dmamux_common_all.h (hunk 1 above)
Comment (grevaillot) [reply include/libopencm3/stm32/common/dmamux_common_all.h]: yeah .. I'm open to any naming suggestion, or clever api refinement.

File: include/libopencm3/stm32/g0/dmamux.h [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+ /** @defgroup dmamux_reg_base DMAMUX register base addresses
+  * @{
+  */
+#define DMAMUX1				DMAMUX_BASE
+/**@}*/
+
+/* --- DMAMUX_CxCR values ------------------------------------ */
+
+/** @defgroup dmamux_cxcr_sync_id SYNCID Synchronization input selected
+  @ingroup dmamux_defines
```
Comment (karlp) [inline include/libopencm3/stm32/g0/dmamux.h]: fyi, you don't need @ingroup markings unless you're trying to put them somewhere else than their logical location.  You're already in dmamux_defines from the defgroup on line 1.

**Comment by karlp** [comment]
Can you fix up the doxygen warnings for me please?

things lik
```
home/karlp/src/libopencm3-tests/libopencm3/include/libopencm3/stm32/common/dmamux_common_all.h:190: warning: The following parameters of dmamux_clear_dma_request_sync_overrun(uint32_t dmamux, uint8_t channel) are not documented:
  parameter 'dmamux'
/home/karlp/src/libopencm3-tests/libopencm3/lib/stm32/common/dmamux.c:353: warning: argument 'dma' of command @param is not found in the argument list of dmamux_clear_request_generator_trigger_overrun_interrupt(uint32_t dmamux, uint8_t rg_channel)
```

have a look in logs doc/stm32g0/doxygen_blah.log

File: include/libopencm3/stm32/g0/dmamux.h (hunk 3 above)
Comment (grevaillot) [reply include/libopencm3/stm32/g0/dmamux.h]: indeed - and you pointed that already in more recent PRs i think ... that bit has been on my repo for too long :)

**Comment by grevaillot** [comment]
done and pushed

**Pull request merged**